### DISCUSSION TITLE
reflection on index fail to match column with key

### DISCUSSION THREAD

**Discussion by holymonson** [sqlalchemy/sqlalchemy#12156] [Usage Questions]
### Describe the bug

In `_reflect_indexes()`, what `_reflect_info` got is the column name in database, while it try to match with the column key. In the case when column `key` and `name` are different, it will fail to match.

https://github.com/sqlalchemy/sqlalchemy/blob/5ded16fae8abfc31d43430cb25757fb434c37ba2/lib/sqlalchemy/engine/reflection.py#L1862-L1871

### SQLAlchemy Version in Use

2.0.36

### DBAPI (i.e. the database driver)

psycopg2

### Database Vendor and Major Version

PostgreSQL 17

### Python Version

3.13

### Operating system

OSX

### To Reproduce

```python
from sqlalchemy import MetaData, Table, Column, Index, INTEGER

metadata = MetaData()
table = Table(
    "table_111",
    metadata,
    Column("col_name", INTEGER, key="col_key"),
    Index("idx_111", Column("col_name")),
)

metadata.reflect(conn, extend_existing=True, autoload_replace=False)
```

### Error

```
>>> metadata.reflect(conn, extend_existing=True, autoload_replace=False)
...
<python-input-17>:1: SAWarning: index key 'col_name' was not located in columns for table 'table_111'
```

**Comment by zzzeek** [comment]
hi -

I'm not sure what you're trying to do above, it's not typically valid to add an `Index` to a table which you are then attempting to run an "extend existing" reflection upon.    extend existing is only meant to be used with extremely minimal tables that have columns only.
